src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -570,6 +575,14 @@ def slice_ramp_data(ramp_data, start_row, nrows):
     ramp_data_slice.flags_saturated = ramp_data.flags_saturated
     ramp_data_slice.flags_no_gain_val = ramp_data.flags_no_gain_val
     ramp_data_slice.flags_unreliable_slope = ramp_data.flags_unreliable_slope
+    ramp_data_slice.flags_chargeloss = ramp_data.flags_chargeloss
+
+    # For possible CHARGELOSS flagging.
+    if ramp_data.orig_gdq is not None:
+        ogdq = ramp_data.orig_gdq[:, :, start_row : start_row + nrows, :].copy()
```

Comment:
The `orig_gdq` is a copy of the group DQ array that comes out of jump, so it's the same shape as a `RampModel` array if `(nints, ngroups, nrows, ncols)`.  Because ramp fitting modifies this DQ array (for example, the CHARGELOSS (and associated DO_NOT_USE) flag is removed during ramp fitting, the original needs to be copied to recompute the read noise variance for ramps with flagged as CHARGELOSS.